### PR TITLE
Added dead letter queue support for the source connector

### PR DESCRIPTION
**Pull request by rozza**
KAFKA-78

https://spruce.mongodb.com/version/5f57982d32f417117277e1e4/tasks

### CODE REVIEW

File: src/main/java/com/mongodb/kafka/connect/source/MongoSourceConfig.java [hunk 1]
```diff
@@ -345,6 +390,131 @@ private static ConfigDef createConfigDef() {
         Width.MEDIUM,
         CONNECTION_URI_DISPLAY);
 
+    configDef.define(
```
Comment (jyemin) [inline src/main/java/com/mongodb/kafka/connect/source/MongoSourceConfig.java:393]: What's the rationale for all the changes to configDef?

File: src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java [hunk 2]
```diff
@@ -255,6 +255,49 @@ public void start(final Map<String, String> props) {
...
+              partition,
+              sourceOffset,
+              topicName,
+              keySchemaAndValue.schema(),
+              keySchemaAndValue.value(),
+              valueSchemaAndValue.schema(),
+              valueSchemaAndValue.value()));
+    } catch (Exception e) {
+      if (sourceConfig.tolerateErrors()) {
+        if (sourceConfig.getBoolean(ERRORS_LOG_ENABLE_CONFIG)) {
```
Comment (jyemin) [inline src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java:281]: Why aren't log levels controlled in the usual way that one does for SLF4J (by configuring log levels in your SLF4J provider config, e.g. logback.xml)

File: src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskTest.java [hunk 3]
```diff
@@ -413,6 +416,52 @@ void testSourceLoadsDataFromCollectionDocumentOnly() {
     }
   }
 
+  @Test
+  @DisplayName("Ensure source sends data to the deadletter queue on failures")
```
Comment (jyemin) [inline src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskTest.java:420]: Is there a test for errorTolerance of ALL and no dead letter queue configured?  If not, let's add one.

**Review by jyemin — CHANGES_REQUESTED** [review]

File: src/main/java/com/mongodb/kafka/connect/source/MongoSourceConfig.java (hunk 1 above)
Comment (rozza) [reply src/main/java/com/mongodb/kafka/connect/source/MongoSourceConfig.java:393]: Purely organizational

File: src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java (hunk 2 above)
Comment (rozza) [reply src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java:281]: This is a Kafka connect configuration that we are borrowing. It allows errors to be ignored and logged and not passed to a DLQ.

File: src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskTest.java (hunk 3 above)
Comment (rozza) [reply src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskTest.java:420]: That will come as part of KAFKA-105

**Review by jyemin — APPROVED** [review]

**Pull request merged**